style.ts

```diff
@@ -48,22 +81,39 @@ export type NegationOperator = '!';
  */
 export type StrMatchesFunctionOperator = 'FN_strMatches';
 
+/**
+ * The Operator used for functional Filters.
+ */
+export type CategorizeFunctionOperator = 'Categorize';
+
 /**
  * All operators.
  */
 export type Operator = ComparisonOperator | CombinationOperator | NegationOperator | StrMatchesFunctionOperator;
 
 /**
- * A FunctionFilter that expects a string (propertyName) as second argument and
- * a regular expression as third argument. An actual parser implementation has to
- * return a value for this function expression.
+ * A FunctionFilter that expects an Expression as second and
+ * third argument. The function checks, if the evaluation of
+ * the second argument matches the evaluation of the third one.
+ * An actual parser implementation has to return a value for
+ * this function expression.
  */
 export type StrMatchesFunctionFilter = [
   StrMatchesFunctionOperator,
-  PropertyName,
+  // TODO just a temporary quick fix
```

Comment:
Is this TODO solved? If not please describe what should be done.

Reply:
No, it's not solved. I'm unsure whether this explicit function type is actually used somewhere or if it can be removed.

Reply:
It is used in the `geostyler-sld-parser`: 
https://github.com/geostyler/geostyler-sld-parser/blob/master/src/SldStyleParser.ts#L333

Reply:
Yes, I'm aware of that. I'm just not sure whether the construct was something like a prototype or is actually used in some project.